laborator/content/operatii-memorie-gdb/7-pointers/pointers.c

```diff
@@ -24,9 +27,51 @@ int main() {
 	cu succes. In caz contrar, programul va crapa.
 	*/
 	// assert(SIGN(my_strcmp(s1, s2)) == SIGN(strcmp(s1, s2)));
-	// assert(my_memcpy(dest, src, sizeof(src)) == memcpy(dest, src, sizeof(src)));
-	// assert(my_strcpy(dest, src) == strcpy(dest, src));
-	free(dest);
+	// assert(test_strcpy(src, sizeof(src)));
+	// assert(test_memcpy(src, sizeof(src)));
 
 	return 0;
 }
+
+
+/* USED FOR TESTING! DO NOT MODIFY! */
+
+int test_strcpy(const char *src, size_t size) {
+	char *dest1 = malloc(size);
+	char *dest2 = malloc(size);
+
+	strcpy(dest1, src);
+	my_strcpy(dest2, src);
+
+	if (strcmp(dest1, dest2)) {
+		free(dest1);
+		free(dest2);
```

Comment:
These `free()`s are duplicated below. Change your logic to only update a variable based on the string comparison and then return that variable after `free()`ing the copied data.